pgb_utils/pgb_utils/utils.py

```diff
@@ -35,7 +36,19 @@ def alert_dict_to_dataframe(alert_dict: dict) -> pd.DataFrame:
 
 def alert_dict_to_table(alert_dict: dict) -> Table:
     """Package a ZTF alert dictionary into an Astopy Table."""
-    rows = [alert_dict["candidate"]] + alert_dict["prv_candidates"]
+    candidate = OrderedDict(alert_dict["candidate"])  # dict
+    cand_cols = set(candidate.keys())
+
+    # astropy 3.2.1 cannot handle dicts with different keys (fixed by v4.1)
+    prv_candidates = []  # recreate with missing keys added
+    for prv_cand in alert_dict["prv_candidates"]:
```

Comment:
Every line inside this loop (except the append at the end) is at least O(n). I think this code block is equivalent to 

```# astropy 3.2.1 cannot handle dicts with different keys (fixed by v4.1)
prv_candidates = []  # recreate with missing keys added
for prv_cand in alert_dict["prv_candidates"]:
    for key in candidate:
        prv_cand.setdefault(key, None)

    prv_candidates.append(prv_cand)
```

which only has a single O(n) call.

Reply:
Thanks for this, it's much cleaner. My only issue with it is that it changes the original alert_dict object. How about using `get()` instead of `setdefault()`?

```python
prv_candidates = []  # recreate with missing keys added
for prv_cand in alert_dict["prv_candidates"]:
    for key in candidate:
        prv_cand.setdefault(key, None)

    prv_candidates.append(prv_cand)
```

Reply:
I'm not clear how using `get` solves the problem. how about using `prv_cand = prv_cand.copy()`?

Reply:
Thanks for this, it's much cleaner. My only issue with it is that it changes the original alert_dict object. How about using `get()` instead of `setdefault()`?

```python
prv_candidates = []  # recreate with missing keys added
for prv_cand in alert_dict["prv_candidates"]:
    prv_cand_tmp = {key: prv_cand.get(key, None) for key in candidate.keys()}
    prv_candidates.append(prv_cand_tmp)
```

Reply:
Sorry, my first comment just copied your code instead of showing mine... 
See my proposed code above. Doesn't that solve the problem?